### PR TITLE
GPT OSS 20B Training Test

### PR DESCRIPTION
**Pull request by pglusacTT**
### Ticket
N/A

### Problem description
We need tests for GPT OSS 20B training.

### What's changed
Added test.

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by vvukomanTT — APPROVED** [review]

File: tests/torch/training/test_moe.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    loss = outputs.loss.mean()
+
+    loss.backward()
+
+    loss_value = loss.item()
+    assert loss_value > 0, f"Loss should be positive, got {loss_value}."
+
+    # Verify gradients exist.
+    assert (
```
Comment (mmilosevicTT) [inline tests/torch/training/test_moe.py]: Shouldn't we check all layers for grads? We had problems with weights tying before in lm_head for llama.

File: tests/torch/training/test_moe.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    shard_specs = build_deinterleaved_shard_specs(model)
+    for tensor, shard_spec in shard_specs.items():
+        xs.mark_sharding(tensor, mesh, shard_spec)
+
+    inputs = model_loader.load_inputs()
+    input_ids = inputs["input_ids"].to(device)
+    attention_mask = inputs["attention_mask"].to(device)
+
+    outputs = model(
+        input_ids=input_ids, attention_mask=attention_mask, labels=input_ids
```
Comment (mmilosevicTT) [inline tests/torch/training/test_moe.py:63]: Should labels be shifted by 1?

File: tests/torch/training/test_moe.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    mesh_shape = (1, num_devices)
+    device_ids = list(range(num_devices))
+    mesh = Mesh(device_ids, mesh_shape, ("batch", "model"))
+
+    device = torch_xla.device()
+
+    model = model.to(device)
+    model.train()
+
+    # Use highest numerical precision for stable fine-tuning convergence.
```
Comment (mmilosevicTT) [inline tests/torch/training/test_moe.py:44]: nit:
```suggestion
    # Use highest numerical precision for stable backward computation.
```

File: tests/torch/training/test_moe.py (hunk 2 above)
Comment (mmilosevicTT) [reply tests/torch/training/test_moe.py:63]: I am just worried this is a no-op pass.

**Review by nsumrakTT — APPROVED** [review]

File: tests/torch/training/test_moe.py (hunk 2 above)
Comment (pglusacTT) [reply tests/torch/training/test_moe.py:63]: It's not a no-op, it executes, I get a loss value. This is just a simplest way of getting any result, this is a smoke test.

**Review by mmilosevicTT — APPROVED** [review]
LGTM

**Review by ndrakulicTT — APPROVED** [review]
LGTM
